### PR TITLE
Use enumer to generate String() methods for most enums

### PR DESCRIPTION
**Pull request by swenson**
We have many hand-written String() methods (and similar) for enums. These require more maintenance and are more error-prone than using automatically generated methods. In addition, the auto-generated versions can be more efficient.

Here, we switch to using https://github.com/loggerhead/enumer, itself a fork of https://github.com/diegostamigni/enumer, no longer maintained, and a fork of the mostly standard tool
https://pkg.go.dev/golang.org/x/tools/cmd/stringer. We use this fork of enumer for Go 1.20+ compatibility and because we require the `-transform` flag to be able to generate constants that match our current code base.

Some enums were not targeted for this change:

* Bit sets, which are not supported by enumer or stringer.
* Enums with non-standard naming schemes, like having a common suffix instead of a prefix, or using spaces instead of underscores or dashes.
* Enums that don't have a type associated

### CODE REVIEW

File: sdk/helper/consts/plugin_runtime_types.go [hunk 1]
```diff
@@ -24,20 +22,11 @@ const (
 	PluginRuntimeTypeContainer
 )
 
-func (r PluginRuntimeType) String() string {
-	switch r {
-	case PluginRuntimeTypeContainer:
-		return "container"
-	default:
-		return "unsupported"
```
Comment (sgmiller) [inline sdk/helper/consts/plugin_runtime_types.go:32]: Curious how enumer handles the defaults?

**Review by sgmiller — COMMENTED** [review]
Neat transform, but my concern is it doesn't preserve the default case for String(), which code may be relying on.

File: sdk/helper/consts/plugin_runtime_types.go (hunk 1 above)
Comment (swenson) [reply sdk/helper/consts/plugin_runtime_types.go:32]: The default will be `TypeName(number)`. For most of these, I tried to ensure that the default wasn't really used, or that the particular `String()` value wasn't relied on.

**Comment by swenson** [comment]
@sgmiller it doesn't preserve the default case we were using, which was lowercase, but where that mattered, I used the `snake` option to match the current `String()` values.

I didn't convert any enums that we couldn't match the current `String()` values for, like ones that have suffixes.

File: builtin/logical/pki/path_tidy.go [hunk 2]
```diff
@@ -23,15 +23,16 @@ import (
 
 var tidyCancelledError = errors.New("tidy operation cancelled")
 
+//go:generate enumer -type=tidyStatusState -trimprefix=tidyStatus
 type tidyStatusState int
 
 const (
-	tidyStatusInactive   tidyStatusState = iota
```
Comment (sgmiller) [inline builtin/logical/pki/path_tidy.go:29]: I guess my other question is whether it's necessary to add full enum functionality to those that were simply used with iota.  Adds code that isn't used...

File: command/healthcheck/healthcheck.go [hunk 3]
```diff
@@ -252,6 +252,7 @@ type Check interface {
 	Evaluate(e *Executor) ([]*Result, error)
 }
 
+//go:generate enumer -type=ResultStatus -trimprefix=Result -transform=snake
```
Comment (sgmiller) [inline command/healthcheck/healthcheck.go:255]: Contrast with this one where we're definitely cleaning up things using enumer.

**Review by sgmiller — APPROVED** [review]

**Comment by swenson** [comment]
Thanks!

**Comment by peteski22** [comment]
if you encountered an error during `make dev` along the lines of:

```
running "enumer": exec: "enumer": executable file not found in $PATH
```

Please run `make tools` (`make bootstrap` likely needs a missing dependency so won't work).

**Pull request merged**